### ISSUE TITLE
Function with too few parameters should not be assignable to variable type

### ISSUE THREAD

**Issue by klausbayrhammer** [microsoft/TypeScript#44986]
# Bug Report

When defining a function type and then applying this function type to a function declaration which declares too few parameters, the typecheck still passes. I would have expected it to fail, since the signatures do not match.

### 🔎 Search Terms

function too few arguments is not assignable to variable type

### 🕗 Version & Regression Information

4.3.5

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/C4TwDgpgBAsiDKBLA5gOwIbAK4CdoF4oAKMdHdAWwEEAuKAZ2B0VWQBopTyKAhOx5qwCUUfAD4oANwD2iACYAoBQGNpqRlAAedOEjSZcBYl0q0GTFsiF0Z80RIDeUAPTOoAdwAWIKHOkR6VAByYChgT2gAFXAIZQjlAGsIHCgAM3REABsoCLwAfgUAXwBuJU0iEIDgIJFXSPgAJgBWJoAWOgBRTUhlYAg5KAaoMmQsCghUYHoOACMsUORpUIBGIA)

### 💻 Code

```ts
type MySignature = (paramA: string, paramB: string) => void

const myFunction: MySignature = (paramA: string): void => { // why doesn't the typechecker fail here?
};

myFunction('test') //TS2554: Expected 2 arguments, but got 1
```

### 🙁 Actual behavior

The assignment of `myFunction: MySignature` passes the typechecker

### 🙂 Expected behavior

I would have expected, that the typechecker fails with an error `Initializer type (paramA: string) => void is not assignable to variable type MySignature`

**Comment by MartinJohns** [comment]
From the FAQ: [Why are functions with fewer parameters assignable to functions that take more parameters?
](https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-functions-with-fewer-parameters-assignable-to-functions-that-take-more-parameters)
